rerun_py/rerun_sdk/rerun/archetypes/image_ext.py

```diff
@@ -228,3 +228,86 @@ def __init__(
             opacity=opacity,
             draw_order=draw_order,
         )
+
+    def compress(self: Any, jpeg_quality: int = 95) -> ImageEncoded | Image:
+        """
+        Compress the given image as a JPEG.
+
+        JPEG compression works best for photographs.
+        Only U8 RGB and grayscale images are supported, not RGBA.
+        Note that compressing to JPEG costs a bit of CPU time,
+        both when logging and later when viewing them.
+
+        Parameters
+        ----------
+        jpeg_quality:
+            Higher quality = larger file size.
+            A quality of 95 saves a lot of space, but is still visually very similar.
+
+        """
+
+        from PIL import Image as PILImage
+
+        from ..archetypes import ImageEncoded
+
+        with catch_and_log_exceptions(context="Image compression"):
+            pixel_format = None
+            if self.pixel_format is not None:
+                pixel_format = PixelFormat(self.pixel_format.as_arrow_array().storage.type_codes[0].as_py())
+
+            # TODO(jleibs): Support conversions here
+            if pixel_format is not None:
+                raise ValueError(f"Cannot JPEG compress an image with pixel_format {pixel_format}")
+
+            color_model = None
+            if self.color_model is not None:
+                color_model = ColorModel(self.color_model.as_arrow_array().storage.type_codes[0].as_py())
+
+            if color_model not in (ColorModel.L, ColorModel.RGB):
+                # TODO(#2340): BGR support!
+                raise ValueError(
+                    f"Cannot JPEG compress an image of type {color_model}. Only L (monochrome) and RGB are supported."
+                )
+
+            datatype = None
+            if self.datatype is not None:
+                datatype = ChannelDatatype(self.datatype.as_arrow_array().storage.type_codes[0].as_py())
+
+            if datatype != ChannelDatatype.U8:
+                # See: https://pillow.readthedocs.io/en/stable/handbook/concepts.html#concept-modes
+                # Note that modes F and I do not support jpeg compression
+                raise ValueError(f"Cannot JPEG compress an image of datatype {datatype}. Only U8 is supported.")
+
+            width = None
+            height = None
+            if self.resolution is not None:
+                resolution = self.resolution.as_arrow_array()[0].as_py()
+                width = resolution[0]
+                height = resolution[1]
+
+            if width is None or height is None:
+                raise ValueError("Cannot JPEG compress an image without a resolution")
+
+            buf = None
+            if self.data is not None:
+                buf = self.data.as_arrow_array().storage.values.to_numpy().view(datatype.to_np_dtype())
```

Comment:
I think it would make sense to have a numpy method like the one you added on tensor [here](https://github.com/rerun-io/rerun/pull/7037/files#diff-757d7ddc6e8814a8f8e21224e57bbe0b0b95514a243664d3701d3f717e6ef1f0R180). Any reason not to?

Reply:
> Any reason not to?

Cross-component dependencies :laughing: :sob: 

Doing this requires knowledge of the datatype, which is stored in a different component. The reason we were able to do that with TensorData is that TensorData is still (for now) both typed, and shaped.

As we split out TensorShape into it's own component, we'll gradually lose the abillity to do that. Perhaps worth thinking more about the data modeling consequences there before we pull that trigger.

Reply:
🤐